### PR TITLE
Rpc/release/clone-components

### PR DESCRIPTION
**Pull request by bliuredhat**
This patch is used to clone release components from one  release to another release.
 The result will return target release.

JIRA: PDC-1204
sub JIRA: PDC-1252

### CODE REVIEW

**Comment by bliuredhat** [comment]
@lubomir ,@ycheng-aa,@simozhan,@

**Comment by bliuredhat** [comment]
Hi,all; please review this patch~!

File: pdc/apps/component/tests.py [hunk 1]
```diff
@@ -1175,6 +1175,19 @@ def test_clone_components_bad_include_inactive_value(self):
...
+        target_response = self.client.post(reverse('release-list'), args)
+        self.assertEqual(target_response.status_code, status.HTTP_201_CREATED)
+        target_release_id = target_response.data['release_id']
+        response = self.client.post(reverse('releasecomponentclone-list'),
+                                    {'source_release_id': 'release-1.0',
+                                     'target_release_id': target_release_id,
+                                     'component_dist_git_branch': 'new_branch'},
+                                    format='json')
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+
```
Comment (simozhan) [inline pdc/apps/component/tests.py]: I think more test cases should be added. For example,the test case for no releasecomponent in source release or there is inactive releasecomponent, etc.

File: pdc/apps/release/views.py [hunk 2]
```diff
@@ -632,6 +632,64 @@ def create(self, request):
...
+        if source_release.releasecomponent_set.count() == 0:
+            return Response(target_data, status=status.HTTP_200_OK)
+
+        signals.rpc_release_clone_component.send(sender=target_release.__class__,
+                                                 request=request,
+                                                 original_release=source_release,
+                                                 release=target_release)
+
+        return Response(target_data, status=status.HTTP_201_CREATED)
+
```
Comment (simozhan) [inline pdc/apps/release/views.py]: Why we return the target release instead of cloned release component?

Comment (bliuredhat) [inline pdc/apps/release/views.py]: I think you are right , but I am not sure what value will return.

**Comment by simozhan** [comment]
:+1:

**Pull request merged**